### PR TITLE
MYEM-3699 added UPDATE and GET subscriber preferences

### PR DESCRIPTION
**Pull request by youcefTab**

### CODE REVIEW

File: notification_lib/novu_manager/subscribers_manager.py [hunk 1]
```diff
@@ -4,7 +4,14 @@
 
 from notification_lib.exceptions import NotificationException
 from notification_lib.http_requests import HttpRequester
-from notification_lib.types import SubscriberCredentials, subscriberPageType, subscriberType
+from notification_lib.types import (
+    SubscriberCredentials,
```
Comment (ouss1919) [inline notification_lib/novu_manager/subscribers_manager.py:8]:
```suggestion
    subscriberCredentials,
```

File: notification_lib/novu_manager/subscribers_manager.py [hunk 2]
```diff
@@ -178,6 +185,66 @@ def update_subscriber_credentials(
...
+    @classmethod
+    def get_subscriber_preferences(cls, subscriber_id: str) -> list[subscriberTemplatePreferencesType]:
+        """Get subscriber preferences for all templates."""
+
+        response = cls.send_request(
+            operation="GET",
+            endpoint=f"/v1/subscribers/{subscriber_id}/preferences",
+        )
+
+        json_response = \
```
Comment (ouss1919) [inline notification_lib/novu_manager/subscribers_manager.py:197]: how this has passed pre-commit ?

File: notification_lib/novu_manager/subscribers_manager.py [hunk 3]
```diff
@@ -178,6 +185,66 @@ def update_subscriber_credentials(
...
+            body=body,
+        )
+
+        json_response = \
+            cls.handle_response(response, "Can't update subscriber preferences for specified template !").json()
+
+        return {
+            "channels": json_response["data"]["preference"]["channels"],
+            "enabled": json_response["data"]["preference"]["enabled"]
+        }
```
Comment (ouss1919) [inline notification_lib/novu_manager/subscribers_manager.py:246]: does it serve to return a data after a patch ?

**Review by ouss1919 — APPROVED** [review]

File: notification_lib/novu_manager/subscribers_manager.py (hunk 2 above)
Comment (youcefTab) [reply notification_lib/novu_manager/subscribers_manager.py:197]: is it about the \ ?

File: notification_lib/novu_manager/subscribers_manager.py (hunk 3 above)
Comment (youcefTab) [reply notification_lib/novu_manager/subscribers_manager.py:246]: we don't particularity need it, but it's a good practice to send back updated data

File: notification_lib/novu_manager/subscribers_manager.py (hunk 2 above)
Comment (ouss1919) [reply notification_lib/novu_manager/subscribers_manager.py:197]: yes

Comment (youcefTab) [reply notification_lib/novu_manager/subscribers_manager.py:197]: it's to mark that what's underneath is on the same line, even blake does sometimes format your code to this format

**Review by houcine-aithammoudi — APPROVED** [review]

**Pull request merged**